### DISCUSSION TITLE
[Feature] Add handling of edited/original files for one image

### DISCUSSION THREAD

**Discussion by justkowal** [immich-app/immich#8169] [Feature Request]
### The feature

As many cloud photo storage apps do, It would be nice for Immich to store unedited copies of photos that user edits (for now in external software, could be extended to built-in editor down the road) and have edited copies act as a main image. Stacks aren't suitable for this because the photo should behave like unedited one unless user specifically requests an unedited copy or undoes changes made to the image. Also, stack lacks structure, it would be hard to determine which stack is a burst, which is a RAW+JPEG and which is Edited/Unedited pair.

What should be added:
- An option that will allow storing edited/original version of an image. In all standard contexts, the main image is the edited one
- An option to get the original, unedited file both via API and the UI
- A button in the mobile app that will allow to open external editor like PicsArt and then save the edit as an edited version on the Immich server.

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Metadata changes are done through sidecar files and so essentially already do this. Whenever we implement image editing it'll definitely work as you describe as we'll never change the original file; we don't need to track that separately, so I'll close this thread.
